### ISSUE TITLE
Slow/intermittent spec failures in full suite, but fast individually

### ISSUE THREAD

**Issue by mjbeauregard** [Quick/Nimble#385]
- [x] I have read [CONTRIBUTING](https://github.com/Quick/Nimble/blob/master/CONTRIBUTING.md) and have done my best to follow them.

### What did you do?

Ran a large test suite with roughly 1600 tests, many of which are async.

### What did you expect to happen?

I'd expect tests to run quickly and consistently every time, whether run individually or part of the full test suite.

### What actually happened instead?

The test suite has recently started taking much longer to run. Some specs take as much as 30 seconds to run within the full test suite, but only 2 seconds if run independently. Also sometimes seeing that there is a stall on main thread.

### Environment

List the software versions you're using:

 - Quick: *0.10.0*
 - Nimble: *5.1.1*
 - Xcode Version: *Version 8.1 (8B62)*
 - Swift Version: `Xcode Default`

Please also mention which package manager you used and its version. Delete the
other package managers in this list:

 - Cocoapods: *1.1.1*

### Project that demonstrates the issue

Unfortunately I'm not able to share the project with the issue.

**Comment by morganchen12** [comment]
If you profile your test suite, is there a lot of time spent in Quick/Nimble calls?

Can you share the trace of the time profiler run?

**Comment by morganchen12** [comment]
If any of your code is setting NSTimers, make sure you're invalidating all of them so they don't stay attached to the main run loop and fire forever while your tests run.

**Comment by jeffh** [comment]
Related to what @morganchen12 says about runloop cruft, you can print out the main loop's [CFRunLoop](https://github.com/Quick/Nimble/issues/177#issuecomment-151393553) to see debugging information about how many timers are actively enqueued.

**Comment by mjbeauregard** [comment]
Here is a followup after considerable debugging:

I managed to trace the slowness down to a particular view controller test. I discovered that due to a retain cycle keeping the window.rootViewController instance alive, there was a condition where the subject under test was trying to present an alert from rootViewController during viewDidLoad. Since the rootViewController was still referring to the previous instance of the subject which had been removed from the view hierarchy in the `afterEach`, the attempt to present the alert was causing UIKit to complain with:

```
Warning: Attempt to present <UIAlertController: 0x7fc3ed7f6110> on <Varo.MoveMoneyViewController: 0x7fc3f08265c0> whose view is not in the window hierarchy!
```

It appears that UIKit can then get into a state where every subsequent view controller test runs very slowly despite the cpu being mostly idle. Fixing the retain cycle so that when the subject is removed from the view hierarchy in `afterEach` cures the warning as well as the slow UIKit behaviour that was slowing the remainder of the test suite.

TLDR; This slowness seemed to be related to a UIKit bug that was triggered by trying to present an alert from a view that is no longer in the view hierarchy.

Thanks for your suggestions and help troubleshooting. I'm closing the issue since it appears to have nothing to do with Quick/Nimble. Thanks!

**Comment by jeffh** [comment]
Thanks for the detailed response Michael! 😃
